version/version.go

```diff
@@ -5,7 +5,7 @@ var TMCoreSemVer = TMVersionDefault
 const (
 	// TMVersionDefault is the used as the fallback version of Tendermint Core
 	// when not using git describe. It is formatted with semantic versioning.
-	TMVersionDefault = "0.34.20"
```

Comment:
When backporting to v0.37.x, should this read `v0.37.0-alpha.1` ?

Reply:
My intuition is to not have this this way in main. I _think_ we'd want the version in `main` to be something like `v0.38.0-dev` or similar to indicate that we are working on v0.38.

Reply:
Agree with `v0.38.0-dev`(I was thinking of doing it later on, by the time we create `v0.38.x`, but can be now as well).
Let me change the base of this PR...

Reply:
... and I'll open another for `main`